mindsdb/api/mysql/mysql_proxy/classes/sql_query.py

```diff
@@ -770,11 +874,39 @@ def execute_step(self, step, steps_data):
         elif type(step) == FetchDataframeStep:
             data = self._fetch_dataframe_step(step)
         elif type(step) == UnionStep:
-            raise ErNotSupportedYet('Union step is not implemented')
-            # TODO add union support
-            # left_data = steps_data[step.left.step_num]
-            # right_data = steps_data[step.right.step_num]
-            # data = left_data + right_data
+            left_result = ResultSet()
+            right_result = ResultSet()
+
+            left_result.from_step_data(steps_data[step.left.step_num])
+            right_result.from_step_data(steps_data[step.right.step_num])
+
+            # count of columns have to match
+            if len(left_result.columns) != len(right_result.columns):
+                raise ErSqlWrongArguments(
+                    f'UNION columns count mismatch: {len(left_result.columns)} != {len(right_result.columns)} ')
+
+            # types have to match
+            for i, left_col in enumerate(left_result.columns):
+                right_col = right_result.columns[i]
+                type1, type2 = left_col.type, right_col.type
+                if type1 is not None and type2 is not None:
```

Comment:
Potentially change to:

```python
if isinstance(type1, str) and isinstance(type2, str):
```

Reply:
But maybe this is a bug in the column type detection, if so, the above would be the wrong fix.